### DISCUSSION TITLE
Chatbot with 3 features

### DISCUSSION THREAD

**Discussion by JonasArmalis** [FlowiseAI/Flowise#749] [General]
Agent or an ability to build a chatbot, which can have
1. Conversational memory with a specific person (userID can be passed in the API call).
2. A knowledge base (can connect a vector store).
3. Prompt. To be able to instruct the agent with a system prompt on how to respond.

The agent should have these 3 essential features. I tried building it, but you can only have zep memory or a vector store connected, there is no agent which has all 3🙂

**Comment by HenryHengZJ** [comment]
you should be able to achieve all 3 using the template - Conversational Retriever Agent: 

<img width="1268" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/26460777/76fc3900-7f09-487b-ba34-a5234eb6ff0e">

just replace the buffer memory to memory like zep, motorhead which have sessionId
